src/components/Feed/PostContainer/PostCaption.tsx

```diff
@@ -1,17 +1,31 @@
-import React from "react";
+import React, { useState } from "react";
 import styles from "./PostContainer.module.css";
 
-const PostCaption: React.FC = () => {
+type PostCaptionProps = {
+  username: string;
+  caption: string;
+};
+
+const PostCaption: React.FC<PostCaptionProps> = ({ caption, username }) => {
+  const [isExpanded, setExpanded] = useState(false);
+  const maxLength = 80;
+
+  function handleExpandClick() {
+    setExpanded(!isExpanded);
```

Comment:
If you want to toggle the previous value, this is not the best way to do it.

There are two alternatives:

1) use previous value `setExpanded(prev => !prev)`
2) wrap your handleExpandClick with `useCallback` and add `isExpanded` as a dependency